Source/DataSources/DataSourceDisplay.js

```diff
@@ -65,9 +67,12 @@ define([
         this._eventHelper = new EventHelper();
         this._eventHelper.add(dataSourceCollection.dataSourceAdded, this._onDataSourceAdded, this);
         this._eventHelper.add(dataSourceCollection.dataSourceRemoved, this._onDataSourceRemoved, this);
+        this._eventHelper.add(dataSourceCollection.dataSourceMoved, this._onDataSourceMoved, this);
 
         this._dataSourceCollection = dataSourceCollection;
         this._scene = scene;
+        this._primitives = scene.primitives.add(new PrimitiveCollection());
```

Comment:
Can we avoid adding these primitives until at least one data source is added?  Otherwise users will always have these two primitives in there app, which is something we try and avoid.

Reply:
That's unfortunate.  Are you sure there's no way for us to create the primitives but not add them until something is actually added to the scene?